src/main/java/service/impl/ValueReportImpl.java

```diff
@@ -0,0 +1,60 @@
+package service.impl;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import service.FileService;
+import service.Strategy;
+import service.Validator;
+import service.ValueReport;
+
+public class ValueReportImpl implements ValueReport {
+    private static final int FIRST_VALUE_LINE = 1;
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_TYPE_LINE_INDEX = 1;
+    private static final int FRUIT_COUNT_LINE_INDEX = 2;
+    private static final String REPORT_CAPTION = "fruit,quantity";
+    private static final String REPORT_SEPARATOR = ",";
+    private final FileService fileService;
+    private final Strategy strategy;
+    private final Validator validator;
+
+    public ValueReportImpl(FileService fileDao, Strategy strategy, Validator validation) {
+        this.fileService = fileDao;
+        this.strategy = strategy;
+        this.validator = validation;
+    }
+
+    public void fillReportMap(List<String> inputValues, Map<String, Integer> report) {
+        validator.validateInformation(inputValues);
+        for (int i = FIRST_VALUE_LINE; i < inputValues.size(); i++) {
+            String[] inputArray = inputValues.get(i).split(REPORT_SEPARATOR);
+            int countFruit = strategy
+                    .getActivity(inputArray[OPERATION_INDEX])
+                    .getFruitAmount(Integer.parseInt(inputArray[FRUIT_COUNT_LINE_INDEX]));
+            report.merge(inputArray[FRUIT_TYPE_LINE_INDEX], countFruit, Integer::sum);
+        }
+    }
```

Comment:
Here you fill database of fruits with data about fruit name and quantity. This class I`d call as Service. It`s service that must only form report. Fill database and get data from it better in separate class. Also better create model TransactionDto, that has fields (Fruit fruit, OperationType(ENUM) or (String) operationType and quantity). Also do this String[] inputArray = inputValues.get(i).split(REPORT_SEPARATOR) in separate parser class (from this you can form TransactionDto).